### PR TITLE
[Teams] Messaging Extension Preview Send/Edit

### PR DESCRIPTION
**Pull request by EricDahlvang**
Add OnTeamsMessagingExtensionBotMessagePreviewSendAsync 
and OnTeamsMessagingExtensionBotMessagePreviewEditAsync 
to ActionBasedMessagingExtensionFetchTaskBot

Fixes: https://github.com/microsoft/botbuilder-dotnet/issues/2619

### CODE REVIEW

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

File: tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            {
+                Question = userText,
+                MultiSelect = choiceSet.IsMultiSelect ? "true" : "false",
+                Option1 = choiceSet.Choices[0].Title,
+                Option2 = choiceSet.Choices[1].Title,
+                Option3 = choiceSet.Choices[2].Title,
+            };
+        }
+
+        public static MessagingExtensionActionResponse GetTaskModuleAdaptiveCardResponse(string userText = null, bool isMultiSelect = true, string option1 = null, string option2 = null, string option3 = null)
```
Comment (johnataylor) [inline tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs]: This seems like an odd function name?
It doesn't "Get" anything, it seems to "Create" a MessagingExtensionActionResponse.

File: tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+using AdaptiveCards;
+using Microsoft.Bot.Schema.Teams;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public static class AdaptiveCardHelper
+    {
+        public static SubmitExampleData GetSubmitExampleDataFromAction(MessagingExtensionAction action)
```
Comment (johnataylor) [inline tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs]: Extension method on MessageExtensionAction? (just an idea to explore?)

File: tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                    {
+                        Type = AdaptiveSubmitAction.TypeName,
+                        Title = "Submit",
+                        Data = new JObject { { "submitLocation", "messagingExtensionFetchTask" } },
+                    },
+                },
+            }.ToTaskModuleResponse();
+        }
+
+        public static AdaptiveCard GetCardFromUserSubmittedData(SubmitExampleData data)
```
Comment (johnataylor) [inline tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs]: Again odd function name?
Possible candidate for an extension function? for example, along the lines submitExampleData.ToAdaptiveCard()

File: tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs (hunk 1 above)
Comment (EricDahlvang) [reply tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs]: Changed to CreateTaskModuleAdaptiveCardResponse

File: tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs (hunk 2 above)
Comment (EricDahlvang) [reply tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs]: Changed to ToSubmitExampleData

File: tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs (hunk 3 above)
Comment (EricDahlvang) [reply tests/Teams/ActionBasedMessagingExtension-FetchTask/AdaptiveCardHelper.cs]: Change to ToAdaptiveCard

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.3.1)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.3.1](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.3.1)

**Pull request merged**